Rakefile

```diff
@@ -1,29 +1,122 @@
-require 'rspec/core/rake_task'
 require 'fileutils'
 require 'rototiller'
+require "yard"
+require "rubocop/rake_task"
+require "flog_task"
+require "flay_task"
+require "roodi_task"
+require "rubycritic/rake_task"
+
+YARD_DIR = "doc".freeze
+DOCS_DIR = "docs".freeze
 
 task :default do
-  sh %{rake -T}
+  sh %(rake -T)
+end
+
+namespace :gem do
+  require "bundler/gem_tasks"
 end
 
-# temporary backwards compat
-task :test => :'test:unit'
-task :acceptance => :'test:acceptance'
+task yard: :"docs:yard"
+
+# rubocop:disable Metrics/BlockLength
+namespace :docs do
+  # docs:yard task
+  YARD::Rake::YardocTask.new
+
+  desc "Clean/remove the generated YARD Documentation cache"
+  task :clean do
+    original_dir = Dir.pwd
+    Dir.chdir(File.expand_path(File.dirname(__FILE__)))
+    sh "rm -rf #{YARD_DIR}"
+    Dir.chdir(original_dir)
+  end
+
+  desc "Tell me about YARD undocummented objects"
+  YARD::Rake::YardocTask.new(:undoc) do |t|
+    t.stats_options = ["--list-undoc"]
+  end
+
+  desc "Measure YARD coverage"
+  require "yardstick/rake/measurement"
+  Yardstick::Rake::Measurement.new(:measure) do |measurement|
+    measurement.output = "yardstick/report.txt"
+  end
+
+  desc "Verify YARD coverage"
+  require "yardstick/rake/verify"
+  config = {"require_exact_threshold" => false}
+  Yardstick::Rake::Verify.new(:verify, config) do |verify|
+    verify.threshold = 57 # FIXME
+  end
+
+  desc "Generate static project architecture graph. (Calls docs:yard)"
+  # this calls `yard graph` so we can"t use the yardoc tasks like above
+  #   We could create a YARD:CLI:Graph object.
+  #   But we still have to send the output to the graphviz processor, etc.
+  task arch: [:yard] do
+    original_dir = Dir.pwd
+    Dir.chdir(File.expand_path(File.dirname(__FILE__)))
```

Comment:
Again, not sure why we need to chdir

Reply:
in this case we might have to `chdir` because yard needs to work on code in a relative path.. and according to the `yard graph` manpage you can't specify a path to analyze.